src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/TypeHelpers.cs

```diff
@@ -347,6 +347,102 @@ internal static DkmClrValue GetNullableValue(this DkmClrValue value, DkmInspecti
             return value.GetFieldValue(InternalWellKnownMemberNames.NullableValue, inspectionContext);
         }
 
+        internal static Type[] GetTupleFieldTypes(this Type type)
+        {
+            var typeArguments = type.GetGenericArguments();
+            int n = typeArguments.Length;
+            Debug.Assert(n > 0);
+            Debug.Assert(n <= TupleFieldRestPosition);
+            if (n < TupleFieldRestPosition)
+            {
+                return typeArguments;
+            }
+            var builder = ArrayBuilder<Type>.GetInstance();
+            for (int i = 0; i < n - 1; i++)
+            {
+                builder.Add(typeArguments[i]);
+            }
+            var restType = typeArguments[n - 1];
+            builder.AddRange(restType.GetTupleFieldTypes());
```

Comment:
>builder.AddRange(restType.GetTupleFieldTypes()); [](start = 12, length = 48)

This recursion allocates arrays we don't need. It looks like we can get away with a loop.